Patient/PatientForm.cs

```diff
@@ -12,9 +14,193 @@ namespace Doctor_Appointment_Management_System.Patient
+                    txtlastname.Text = reader["last_name"].ToString();
+                    txtdob.Text = reader["dob"].ToString();
+                    txtaddress.Text = reader["address"].ToString();
+                    txtphone.Text = reader["phone_number"].ToString();
+                }
+            }
+
+            // we do not need the connection any more, close the database connection
+            Databse.DatabaseConnection.close();
+        }
+
+       /* private void button1_Click(object sender, EventArgs e)
+        {
+            loadPatient(txtloadpatientId.Text);
+        }*/
+
+
+        private void updatePatient()
+        {
+            if (isValidPatientForm())
+            {
+                try
+                {
+                    SqlCommand patientInsertCommand;
+
+                    patientInsertCommand = new SqlCommand("UPDATE Patient SET first_name=@firstname,last_name=@lastname,dob=@dob,address=@address,phone_number=@phone where patient_id=@id", this.databaseConnection);
+                    Databse.DatabaseConnection.open(); // open databse
+
+                    // bind values to insert query
+                    patientInsertCommand.Parameters.AddWithValue("@id", txtpatientid.Text);
+                    patientInsertCommand.Parameters.AddWithValue("@firstname", txtfirstname.Text);
+                    patientInsertCommand.Parameters.AddWithValue("@lastname", txtlastname.Text);
+                    patientInsertCommand.Parameters.AddWithValue("@dob", txtdob.Text);
+                    patientInsertCommand.Parameters.AddWithValue("@address", txtaddress.Text);
+                    patientInsertCommand.Parameters.AddWithValue("@phone", txtphone.Text);
+
+                    // execute the insert command, data will be insert into database
+                    patientInsertCommand.ExecuteNonQuery();
+
+                    // we do not need the connection any more close the database connection
+                    Databse.DatabaseConnection.close(); ;
+
+                    // show success message to user
+                    MessageBox.Show("Patient updated successfully!");
+                }
+                catch (Exception ex)
+                {
+                    MessageBox.Show("Patient update failed: " + ex.Message);
+                }
+            }
+        }
+
+        private void savePatient()
+        {
+            if (isValidPatientForm())
+            {
+                try
+                {
+                    SqlCommand patientInsertCommand;
+
+                    patientInsertCommand = new SqlCommand("INSERT INTO Patient (patient_id,first_name,last_name,dob,address,phone_number) VALUES (@patientid,@firstname,@lastname,@dob,@address,@phone)", this.databaseConnection);
+                    Databse.DatabaseConnection.open(); // open databse
+
+                    // bind values to insert query
+                    patientInsertCommand.Parameters.AddWithValue("@patientid", txtpatientid.Text);
```

Comment:
Use camelCase for naming variables
txtfirstname = > txtFirstName
txtpatientid => txtPatientId

etc